src/builtin_string.cpp

```diff
@@ -477,7 +481,7 @@ class pcre2_matcher_t: public string_matcher_t
         {
```

Comment:
@floam: If I add `if (opts.invert_match) return 0;` directly above this loop it seems to fix the return value for `-rv`, and my internal logic also says that should be okay - if we have a match, we want to ignore it. We already added any non-match above.

The below `if (!opts.quiet && !opts.invert_match)` can probably be reverted to `if (!opts.quiet)` then.